### DISCUSSION TITLE
MX redundancy

### DISCUSSION THREAD

**Discussion by savchenko** [emersion/maddy#473] [Q&A]
Is there any sort of FAQ on the recommended way to setup a spare MX with Maddy?

Imagine a user with Maddy installed on two servers, `mx1` and `mx2`:
```
10 mx1.example.com 3600
20 mx2.example.com 3600
```

`mx1` goes down and 3rd-party MTA delivers to `mx2`. How to keep these two in-sync?

**Comment by cbrake** [comment]
There are some neat things going on with sqlite that may eventually be a replication solution:

https://twitter.com/benbjohnson/status/1494800752333647872

**Reply by savchenko** [reply]
Reading the https://github.com/foxcpp/maddy/discussions/463 it seems like SQLite might be suboptimal for the purpose, no?

**Reply by cbrake** [reply]
that is a good question. From what I hear, sqlite is quite fast and robust, so it seems it could be made to work for a small business mail server store. However, I don't know much about the db tradeoffs yet.

**Comment by angelnu** [comment]
as reference - I am running successfully with a Helm chart I created at https://github.com/angelnu/helm-charts/tree/main/charts/apps/maddy

I use it together with a Zalando Postgres cluster for HA: https://github.com/angelnu/k8s-gitops/blob/main/core/maddy/release.yaml

As my chart is based on a [library chart](https://bjw-s.github.io/helm-charts/docs/common-library/introduction/) that allows for defining the number of instances I tried to start 2 maddy replicas and it seems to work fine. So beyond some optimizations and the SMTP queue being local (which is fine in my case as I send via sendmail) it is possible to run some form of #279
